aws/signer/v4/v4.go

```diff
@@ -678,6 +693,10 @@ func (ctx *signingCtx) isRequestSigned() bool {
 	return false
 }
 
+func (ctx *signingCtx) sanitizeHostForHeader() {
+	aws.SanitizeHostForHeader(ctx.Request)
```

Comment:
The `sanitizeHostForHeader` method on `signingCtx` seems redundant could the `aws.SanitizeHostForHeader` be called on line 332 instead of this method?